### PR TITLE
Fix navigation to channels using mention links

### PR DESCRIPTION
**Pull request by mrsimpson**
When clicking a channel link (created either using a thread or by simply mentioning it using `#channel`), the link was not resolved if the user was not a member of the target channel.

### CODE REVIEW

File: packages/rocketchat-ui/client/views/app/room.js [hunk 1]
```diff
@@ -730,6 +729,17 @@ Template.room.events({
 				}
 
 				FlowRouter.goToRoomById(room._id);
```
Comment (vickyokrm) [inline packages/rocketchat-ui/client/views/app/room.js]: ``goToRoomByID`` does not resolve when there is no subscription exists for the user. The user is not a member of a group and he/she not allowed to navigate to the target channel(Both thread and mention-link).

Comment (mrsimpson) [reply packages/rocketchat-ui/client/views/app/room.js]: Exactly. This is why I implemented the else tarting from line 732.
The room in line 726 will only be assigned also if a subscription exists for the user.
Can you tell me which navigation does not work?

Comment (vickyokrm) [reply packages/rocketchat-ui/client/views/app/room.js]: User created a new thread from 'General' channel. Later he was removed as owner and kicked out of the room.Now when the user tries again to access the link. it works for the first time but not the subsequent clicks.

Comment (vickyokrm) [reply packages/rocketchat-ui/client/views/app/room.js]: Navigation works as excepted.
What is tested
1) Public Channel:
   - Navigation between public should be possible even-though the user is not member of the target channel
2) Private Channel:
    - Navigation between private should be not possible if the user is not member of the target channel. 

What is missing: 
Show the **Not Authorised** Page if the user do not have access to the target channel.

**Comment by mrsimpson** [comment]
True! Can you create a big for what’s missing? This is something which does not work in plain RC either, so I’d consider this a separate bug

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by vickyokrm** [comment]
Reported the issue to RC [12023](https://github.com/RocketChat/Rocket.Chat/issues/12023).

**Comment by vickyokrm** [comment]
Re-running tests to make to pass. will merge once it succeed.

**Pull request merged**